src/ray/object_manager/plasma/object_lifecycle_manager.h

```diff
@@ -0,0 +1,152 @@
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#pragma once
+
+#include "absl/container/flat_hash_set.h"
+#include "gtest/gtest.h"
+#include "ray/object_manager/plasma/common.h"
+#include "ray/object_manager/plasma/eviction_policy.h"
+#include "ray/object_manager/plasma/object_store.h"
+#include "ray/object_manager/plasma/plasma_allocator.h"
+
+namespace plasma {
+
+// ObjectLifecycleManager allocates LocalObjects from the allocator.
+// It tracks object’s lifecycle states including reference_count and created/sealed.
+// It uses eviction_policy to garbage collect objects when running out of space.
+class ObjectLifecycleManager {
+ public:
+  ObjectLifecycleManager(IAllocator &allocator,
+                         ray::DeleteObjectCallback delete_object_callback);
+
+  /// Create a new object given object's info. Object creation might
+  /// fail if runs out of space; or an object with the same id exists.
+  ///
+  /// \param object_info Plasma object info.
+  /// \param source From where the object is created.
+  /// \param fallback_allocator Whether to allow fallback allocation.
+  /// \return
+  ///   - pointer to created object and PlasmaError::OK when succeeds.
+  ///   - nullptr and error message, including ObjectExists/OutOfMemory
+  std::pair<const LocalObject *, flatbuf::PlasmaError> CreateObject(
+      const ray::ObjectInfo &object_info, plasma::flatbuf::ObjectSource source,
+      bool fallback_allocator);
+
+  /// Get object by id.
+  /// \return
+  ///   - nullptr if such object doesn't exist.
+  ///   - otherwise, pointer to the object.
+  const LocalObject *GetObject(const ObjectID &object_id) const;
+
+  /// Seal created object by id.
+  ///
+  /// \param object_id Object ID of the object to be sealed.
+  /// \return
+  ///   - nulltpr if such object doesn't exist, or the object has already been sealed.
+  ///   - otherise, pointer to the sealed object.
+  const LocalObject *SealObject(const ObjectID &object_id);
+
+  /// Abort object creation by id. It deletes the object regardless of reference
+  /// counting.
+  ///
+  /// \param object_id Object ID of the object to be aborted.
+  /// \return
+  ///   - false if such object doesn't exist, or the object has already been sealed.
+  ///   - true if abort successfuly.
+  bool AbortObject(const ObjectID &object_id);
+
+  /// Delete a specific object by object_id. The object is delete immediately
+  /// if it's been sealed and reference counting is zero. Otherwise it will not
```

Comment:
`Otherwise it will not deleted` seems to be confusing. Can you mention it will be deleted once there's no usage?